src/App/Pages/Accounts/LoginSsoPageViewModel.cs

```diff
@@ -235,18 +230,53 @@ private async Task LogIn(string code, string codeVerifier, string orgId)
                         _syncService.FullSyncAsync(true).FireAndForget();
                         SsoAuthSuccessAction?.Invoke();
                     }
+                    else if (pendingRequest != null)
+                    {
+                        var authRequest = await _authService.GetPasswordlessLoginRequestByIdAsync(pendingRequest.Id);
+                        if (authRequest != null && authRequest.RequestApproved != null && authRequest.RequestApproved.Value)
+                        {
+                            var authResult = await _authService.LogInPasswordlessAsync(await _stateService.GetActiveUserEmailAsync(), authRequest.RequestAccessCode, pendingRequest.Id, pendingRequest.PrivateKey, authRequest.Key, authRequest.MasterPasswordHash);
+                            if (authResult == null && await _stateService.IsAuthenticatedAsync())
+                            {
+                                await Xamarin.Essentials.MainThread.InvokeOnMainThreadAsync(
+                                 () => _platformUtilsService.ShowToast("info", null, AppResources.LoginApproved));
+                                await _stateService.SetPendingAdminAuthRequestAsync(null);
+                                _syncService.FullSyncAsync(true).FireAndForget();
+                                SsoAuthSuccessAction?.Invoke();
+                            }
+                        }
+                        else
+                        {
+                            StartDeviceApprovalOptionsAction?.Invoke();
+                        }
+                    }
                     else
                     {
                         StartDeviceApprovalOptionsAction?.Invoke();
                     }
+                    return;
                 }
-                else
+
+                // In the standard, non TDE case, a user must set password if they don't
+                // have one and they aren't using key connector.
+                // Note: TDE & Key connector are mutually exclusive org config options.
+                var requireSetPassword = !decryptOptions?.HasMasterPassword ?? false && decryptOptions?.KeyConnectorOption == null;
```

Comment:
❓ Is there a possibility where `decryptOptions` is `null` but the user is using `Key connector`? Or that can't happen on login with the new changes?

⛏️ Given that the information expert is the `DecryptOptions` object, can this be moved inside the `DecryptOptions` with a property `RequireSetPassword`? so then you can directly use `decryptOptions?.RequireSetPassword`

Reply:
Yes that is a possibility, I forgot we also need backwards compatibility for key connector, `tokenResponse.KeyConnectorUrl`. I also don't see the `_keyConnectorService.GetAndSetKey()` call, am I just missing that?

Reply:
Fixed by adding code for new user sso in auth service